src/main/java/com/project/trackfit/core/ApplicationUserController.java

```diff
@@ -22,13 +21,13 @@
 @AllArgsConstructor
 @RequestMapping("/api/v1/auth/register")
 public class ApplicationUserController {
-    private  final ApplicationUserService applicationUserService;
+
+    private final ApplicationUserService applicationUserService;
+
     @PostMapping
-    public ResponseEntity<APICustomResponse>createUser(
-            @Valid
-            @RequestBody
-            CreateUserRequest createUserRequest) throws NoSuchAlgorithmException {
-        UUID userId=applicationUserService.createUser(createUserRequest);
+    public ResponseEntity<APICustomResponse> createUser(@Valid @RequestBody CreateUserRequest createUserRequest)
+            throws NoSuchAlgorithmException {
+        UUID userId = applicationUserService.createUser(createUserRequest);
```

Comment:
why NoSuchAlgorithmException

Reply:
This exception has to be handled when using the following code snippet: ``md = MessageDigest.getInstance("SHA-512");``

To resolve this issue we added a try / catch block and removed the ``throws NoSuchAlgorithmException`` from every other method declaration.